clang/test/lab1/vyunov_danila/always_inline_attr.cpp

```diff
@@ -0,0 +1,139 @@
+// RUN: %clang_cc1 -ast-dump -load %llvmshlibdir/vyunovAlwaysInlinePlugin%pluginext -add-plugin AddAlwaysInline %s 1>&1 | FileCheck %s --check-prefix=SUM
+// SUM: __attribute__((always_inline)) int sum(int A, int B) {
+// SUM-NEXT:   return A + B;
+// SUM-NEXT: }
+// SUM: `-AlwaysInlineAttr {{.* Implicit always_inline}}
+int sum(int A, int B) { return A + B; }
+
+// RUN: %clang_cc1 -load %llvmshlibdir/vyunovAlwaysInlinePlugin%pluginext -plugin AddAlwaysInline %s 1>&1 | FileCheck %s --check-prefix=EMPTY
+// EMPTY: __attribute__((always_inline)) void checkEmpty() {
+// EMPTY-NEXT: }
+void checkEmpty() {}
+
+// RUN: %clang_cc1 -load %llvmshlibdir/vyunovAlwaysInlinePlugin%pluginext -plugin AddAlwaysInline %s 1>&1 | FileCheck %s --check-prefix=MIN-NESTED
+// MIN-NESTED: int minNested(int A, int B) {
+// MIN-NESTED-NEXT:     {
+// MIN-NESTED-NEXT:         if (A < B) {
+// MIN-NESTED-NEXT:             return A;
+// MIN-NESTED-NEXT:         }
+// MIN-NESTED-NEXT:         return B;
+// MIN-NESTED-NEXT:     }
+// MIN-NESTED-NEXT: }
```

Comment:
Expected checking of a negative case:
```suggestion
// MIN-NESTED: `-FunctionDecl {{.*}} minNested
// MIN-NESTED-NOT: AlwaysInlineAttr
```